modernomad/settings/staging.py

```diff
@@ -2,3 +2,6 @@
 from .common import env
 
 SECRET_KEY = env('SECRET_KEY')
+ANYMAIL_SEND_DEFAULTS = {
+    'esp_extra': {'o:testmode': 'yes'}
```

Comment:
This I'd say is probably the most crucial bit. See https://anymail.readthedocs.io/en/stable/esps/mailgun/#exp-extra-support

So with these settings testmode is activated for staging provided that staging settings are being used on staging. Perhaps you can confirm that @bfirsh? 

On local the console backend is being used so no mail is ever passed onto mailgun in the first place.